### PR TITLE
[FEAT] Add gradient_runtime dependency, remove type annotation file

### PR DESCRIPTION
**Pull request by luk-pau-es**
Replace `Gradient.TypeAnnotation` file with dependency `gradient_runtime`

Resolves Enhancement [#147](https://github.com/esl/gradient/issues/147)

Gradient Runtime [repo](https://github.com/esl/gradient_runtime)

### CODE REVIEW

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -9,7 +9,7 @@ on:
 
 jobs:
   test:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-20.04
```
Comment (luk-pau-es) [inline .github/workflows/build-and-test.yml:12]: I needed to downgrade ubuntu version from ubuntu-latest (22.04 as of the moment of writing the comment) to lower: 20.04 as ubuntu 22 is not compatible with Erlang 23 as per compatibility table here: https://github.com/erlef/setup-beam#compatibility-between-operating-system-and-erlangotp

File: mix.exs [hunk 2]
```diff
@@ -30,6 +30,7 @@ defmodule Gradient.MixProject do
   def deps do
     [
       {:gradualizer, github: "josefs/Gradualizer", ref: "5567a23", manager: :rebar3},
+      {:gradient_runtime, github: "esl/gradient_runtime", only: [:test, :prod], ref: "ee8de90"},
```
Comment (erszcz) [inline mix.exs:33]: Why would we only rely on this in `test` and `prod`, but not `dev`?

**Comment by erszcz** [comment]
It seems to me I've too hastily triaged #147 and that we don't really need a separate package. @luk-pau-es could you take a look at https://github.com/esl/gradient/issues/147#issuecomment-1350656340 and see if you agree that's the case? If it is, then we can close this PR without merging. Sorry for the confusion.

**Comment by luk-pau-es** [comment]
I have read through the comment on the task, and I agree with you. I am closing this PR.

**Comment by erszcz** [comment]
Fix #147

**Pull request merged**